web/modules/webspark/webspark_blocks/js/card-arrangement.js

```diff
@@ -0,0 +1,71 @@
+(function ($, Drupal, drupalSettings, once) {
+  // Define a behavior for handling card arrangement alerts
+  Drupal.behaviors.cardArrangementAlert = {
+    attach: function (context, settings) {
+      // Initialize variables
+      const selectCountColumn = document.querySelector('[name="settings[block_form][field_card_arrangement_display]"]');
+      const displayOrientation = document.querySelector('[name="settings[block_form][field_display_orientation]"]');
+      const displayDiv = document.querySelector('[data-drupal-selector="edit-settings-block-form-field-display-orientation-wrapper"]');
+
+      // Get the element to identify Ranking cards
+      const elementRanking = document.querySelector('[name*="[field_card_ranking_image_size]"]');
+      // Get the element to identify Image based cards
+      const elementBasedCard = document.querySelector('[name*="[field_loading]"]');
+
+      // Check if the element exists to determine display orientation
+      if (elementRanking || elementBasedCard) {
+        // Hide display orientation
+        _showDisplayOrientation('none');
+      } else {
+        // Show display orientation
+        _showDisplayOrientation();
+      }
```

Comment:
This seems to be working fine, but I'm wondering if you considered using the [Drupal Conditional Form States API](https://www.drupal.org/docs/drupal-apis/form-api/conditional-form-fields) to manage the form field logic? I believe everything this script does could be implemented using that API.

Example of this being used in the `asu_degree_rfi` module is at https://github.com/ASUWebPlatforms/webspark-ci/blob/master/web/modules/webspark/asu_degree_rfi/asu_degree_rfi.module#L90-L110

Reply:
This comment is referring to the whole card-arrangement.js file.

Reply:
Thank you for checking! I just wanted to make sure we weren't "reinventing the wheel" but sounds like our custom code is required. :)